File: hello.java
import java.util.ArrayList;
import java.util.List;

class HelloWorld {
  public static void main(String[] args) {
    int count = Integer.parseInt(System.getenv("COUNT"));
    List<Integer> primes = fillPrimesTil(count);
    System.out.println(String.format("%d primes between 1 and %d", primes.size(), count));
  }

  private static List<Integer> fillPrimesTil(int i) {
    List<Integer> l = new ArrayList<Integer>(1);
    for (int count = 0; count < i; count++) {
      if (isPrime(count)) {
        l.add(count);
      }
    }
    return l;
  }

  private static boolean isPrime(int number) {
    if (number <= 1) return false;
    for (int i = 2; i < number; i++) {
      if (number % i == 0) return false;
    }
    return true;
  }
}

File: hello.kt
fun main() {
    val count = System.getenv("COUNT").toInt()
    val primes = fillPrimesTil(count)
    println("${primes.size} primes between 1 and $count")
}

fun fillPrimesTil(i: Int): List<Int> {
    val l = mutableListOf<Int>()
    for (count in 0 until i) {
        if (isPrime(count)) {
            l.add(count)
        }
    }
    return l
}

fun isPrime(number: Int): Boolean {
    if (number <= 1) return false
    for (i in 2 until number) {
        if (number % i == 0) return false
    }
    return true
}